scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2154,7 +2154,7 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
 
   private def exprMaybeIndented(): Term = {
     if (token.is[Indentation.Indent]) {
-      block()
+      blockExpr()
```

Comment:
Funnily enough this also allows:
```
if
   case a =>
   case b =>
then
  stat
```

which is not entirely correct, but it's also accepted by the Dotty parser, only to be rejected at typer. I think more permissive parser is ok in this case.

Reply:
I agree it's better to be permissive in this case. I think the Scalameta parser is probably too strict in some cases.